pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,13 +616,30 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
+            if len(self._obs_data()) != len(other._obs_data()):
```

Comment:
Make `# pylint: disable=protected-access` the first line of this method to fix one of the pylint warnings.